### ISSUE TITLE
Incorrect focus theme used for FormFieldContentBox

### ISSUE THREAD

**Issue by tstanev** [grommet/grommet#4210]
Form field use incorrect border color when in focus, when the them specifies a `formField.focus.border.color` and a global focus color is also set. This was likely caused by changes in https://github.com/grommet/grommet/pull/4018, where FormFieldContentBox was modified to get the global focus theme and ignore the evaluated border color passed in by the parent component.

### Expected Behavior

Expected that form fields use the theme from `formField.focus.border.color`

### Actual Behavior

Form fields override `formField.focus.border.color` with the global `focus.border.color` setting.

It appears that this code added in the mentioned PR :
```
const FormFieldContentBox = styled(Box)`
  ${props => props.focus && focusStyle({ justBorder: true })}
`;
```
is overriding the formField specific borderColor from props (which is correct) by the global focus color obtained by focusStyle.

### Your Environment

Grommet version 2.14, Chrome 83 (though browser shouldn't matter in this case)

**Comment by tstanev** [comment]
@ericsoderberghp As you are the author of that PR, would appreciate your input about this and whether it's the intended behavior to override the `borderColor` coming in from `innerProps` into FormFieldContentBox.

**Comment by ericsoderberghp** [comment]
Good question. The overall intent was to ensure that the type of focus styling could be controlled by the theme and be applied consistently across components that can receive focus. Just so I can understand the context of what you are wanting to do better, can you elaborate on what you are trying to accomplish? For example, are you wanting the global focus to appear one way, perhaps in style or color, and the focus in FormField to appear a different way, perhaps in a different style or color? If so, what is driving the desire for inconsistent focus behavior. I think accessibility is generally better if the focus indicator is consistent across components. But, perhaps I am not understanding the context well enough.

**Comment by tstanev** [comment]
This is roughly what we are doing: we have a global focus color set to transparent to avoid focus rectangle from a bunch of elements that don't need it, and we want to specifically have a non-transparent focus color for form fields, where it's important.
From the code it looked like it's making an attempt to pass the formField focus color down to FormFieldContentBox, but then ignores it and takes the global focus color instead.
So basically the question is, is it considered correct behavior that the global focus color takes precedence over the specific component theme's focus color?
